### PR TITLE
 [enterprise-4.14] TELCODOCS-1838 Update reference SiteConfig to use rootDeviceHints with consistent naming

### PR DESCRIPTION
**Pull request by kquinn1204**
[TELCODOCS-1838]: Docs: Create partitions through SiteConfig

Version(s):
4.14

Issue:https://issues.redhat.com/browse/TELCODOCS-1832

Link to docs preview:
https://79811--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.html
https://79811--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-sites.html
https://79811--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-precaching-tool.html

QE review:
- [x] QE has approved this change.

Additional information:
“Cherry Picked from d59e7cf02bd625fb56cd46e16d91b48c6859a07b xref: [https://github.com/openshift/openshift-docs/pull/75726]

### CODE REVIEW

File: snippets/ztp_example-sno.yaml [hunk 1]
```diff
@@ -12,95 +12,140 @@ spec:
...
+        # disk partition at `/var/lib/containers` with ignitionConfigOverride. Some values must be updated. See DiskPartitionContainer.md for more details
+        ignitionConfigOverride: | 
+           {
+            "ignition": {
+              "version": "3.2.0"
+            },
+            "storage": {
+              "disks": [
+                {
+                  "device": "/dev/disk/by-id/wwn-0x6b07b250ebb9d0002a33509f24af1f62",
```
Comment (sabbir-47) [inline snippets/ztp_example-sno.yaml:87]: please change the devices by-id to by-path like below
`"device": "/dev/disk/by-path/pci-0000:01:00.0-scsi-0:2:0:0"`

**Comment by sabbir-47** [comment]
I have put one comment. Otherwise lgtm.

File: modules/ztp-configuring-disk-partitioning.adoc [hunk 2]
```diff
@@ -8,39 +8,197 @@
...
 
 .Prerequisites
 
-* You have installed the OpenShift CLI (`oc`).
+* Install Butane.
+
+.Procedure
 
-* You have logged in to the hub cluster as a user with `cluster-admin` privileges.
+. Create the `storage.bu` file.
```
Comment (lpettyjo) [inline modules/ztp-configuring-disk-partitioning.adoc:22]: Would recommend that you add something like: "...by using the following example YAML file:"

File: modules/ztp-configuring-disk-partitioning.adoc [hunk 3]
```diff
@@ -8,39 +8,197 @@
...
+.Example output
+[source,terminal]
+----
+"{\"ignition\":{\"version\":\"3.2.0\"},\"storage\":{\"disks\":[{\"device\":\"/dev/disk/by-id/wwn-0x6b07b250ebb9d0002a33509f24af1f62\",\"partitions\":[{\"label\":\"var-lib-containers\",\"sizeMiB\":0,\"startMiB\":250000}],\"wipeTable\":false}],\"filesystems\":[{\"device\":\"/dev/disk/by-partlabel/var-lib-containers\",\"format\":\"xfs\",\"mountOptions\":[\"defaults\",\"prjquota\"],\"path\":\"/var/lib/containers\",\"wipeFilesystem\":true}]},\"systemd\":{\"units\":[{\"contents\":\"# Generated by Butane\\n[Unit]\\nRequires=systemd-fsck@dev-disk-by\\\\x2dpartlabel-var\\\\x2dlib\\\\x2dcontainers.service\\nAfter=systemd-fsck@dev-disk-by\\\\x2dpartlabel-var\\\\x2dlib\\\\x2dcontainers.service\\n\\n[Mount]\\nWhere=/var/lib/containers\\nWhat=/dev/disk/by-partlabel/var-lib-containers\\nType=xfs\\nOptions=defaults,prjquota\\n\\n[Install]\\nRequiredBy=local-fs.target\",\"enabled\":true,\"name\":\"var-lib-containers.mount\"}]}}"
+----
+--
+
+. After installation, check the {sno} disk status. 
+
+.. Enter into a debug session on the {sno} node by running the following command. This step instantiates a debug pod called `<node_name>-debug`:
```
Comment (lpettyjo) [inline modules/ztp-configuring-disk-partitioning.adoc:145]: Recommend you move the explanatory statement to the next line. Use a + sign between the step action statement and the explanatory statement to keep them on the same indent.

File: modules/ztp-configuring-disk-partitioning.adoc [hunk 4]
```diff
@@ -8,39 +8,197 @@
...
++
+--
+[source,terminal]
+----
+$ oc debug node/my-sno-node
+----
+
+.. Set `/host` as the root directory within the debug shell by running the following command. The debug pod mounts the host’s root file system in `/host` within the pod. By changing the root directory to `/host`, you can run binaries contained in the host’s executable paths:
```
Comment (lpettyjo) [inline modules/ztp-configuring-disk-partitioning.adoc:153]: Same comment as above.

File: snippets/ztp_example-sno.yaml [hunk 5]
```diff
@@ -12,95 +12,140 @@ spec:
...
+      {
+        "capabilities": {
+          "baselineCapabilitySet": "None",
+          "additionalEnabledCapabilities": [
+            "NodeTuning",
+            "OperatorLifecycleManager"
+          ]
+        }
+      }
+    # It is strongly recommended to include crun manifests as part of the additional install-time manifests for 4.13+.
```
Comment (lpettyjo) [inline snippets/ztp_example-sno.yaml:35]: What does 4.13+ refer to? I assume you mean OpenShift. If so, state explicitly and also use the convention of "{product-title} 4.13 or later."

**Review by lpettyjo — COMMENTED** [review]
Otherwise, LGTM!

File: snippets/ztp_example-sno.yaml (hunk 5 above)
Comment (kquinn1204) [reply snippets/ztp_example-sno.yaml:35]: This reflects a YAML file created by dev so I cannot change this as it needs to be the same as one downloaded as part of zip file.

**Review by sheriff-rh — APPROVED** [review]
Changes look good; merging.

**Pull request merged**